website/static/js/citationsFolderPickerViewModel.js

```diff
@@ -0,0 +1,487 @@
+var ko = require('knockout');
+require('knockout-punches');
+var $ = require('jquery');
+var bootbox = require('bootbox');
+var Raven = require('raven-js');
+
+var FolderPicker = require('folderpicker');
+var $osf = require('osfHelpers');
+var ctx = window.contextVars;
+
+ko.punches.enableAll();
+/**
+ * Knockout view model for citations node settings widget.
+ */
+var CitationsFolderPickerViewModel = function(name, url, selector, folderPicker) {
+    var self = this;
+
+    self.url = url;
+    self.properName = name.charAt(0).toUpperCase() + name.slice(1);
+    self.selector = selector;
+    // CSS selector for the folder picker div
+    self.folderPicker = folderPicker;
+
+    // DEFAULTS
+    self.accounts = ko.observableArray([]);
+    // Auth information
+    self.nodeHasAuth = ko.observable(false);
+    // whether current user is authorizer of the addon
+    self.userIsOwner = ko.observable(false);
+    // whether current user has an auth token
+    self.userHasAuth = ko.observable(false);
+    // whether the auth token is valid
+    self.validCredentials = ko.observable(true);
+    self.userAccountId = ko.observable('');
+    // Currently linked folder, an Object of the form {name: ..., id: ...}
+    self.folder = ko.observable({
+        name: '',
+        id: ''
+    });
+    self.ownerName = ko.observable('');
+    self.urls = ko.observable({});
+    // Flashed messages
+    self.message = ko.observable('');
+    self.messageClass = ko.observable('text-info');
+    // Display names
+    self.PICKER = 'picker';
+    // Current folder display
+    self.currentDisplay = ko.observable(false);
+    // Currently selected folder name
+    self.selected = ko.observable(false);
+    self.loading = ko.observable(false);
+    // Whether the initial data has been fetched form the server. Used for
+    // error handling.
+    self.loadedSettings = ko.observable(false);
+    // Whether the mendeley folders have been loaded from the server/mendeley API
+    self.loadedFolders = ko.observable(false);
+
+    self.messages = {
+        INVALID_CRED_OWNER: ko.computed(function() {
```

Comment:
All of these can probably be changed to `ko.pureComputed`. @samchrisinger Can you make this change post-merge?